### PR TITLE
Routing Additions: recipes and ingredients

### LINKED ISSUE #10: recipes/<username> Route
The recipes/<username> route should return JSON information of: id, name, time, ingredients, link of all recipies which the username posesses at least 1 of the ingredients in the recipe.

Should do this by passing username into database service which returns the data.

### LINKED ISSUE #11: ingredients/<username> route
New route which returns all individual ingredients of a specific user from the database. 
In JSON format returns id and ingredient name.

Should do this by runnning a database service which takes in the username and produces the ids and ingredient names.

### PR DESCRIPTION
**Pull request by Deevun155**
add routes for /ingredients/username and /recipes/username

also edit houseingre function to output both the ingredient name and id for each ingredient, which necessitated the need to change the query function to only pass the ingredient names into the houserecipes function

resolves #10 #11

@StewartSmyth Please check that the data in the JSON is of the correct form to be used in the frontend, and @Zev0l3 please check that the changes made to service.py won't mess anything else up that relates to the database functions. I'll fix if needed.

### CODE REVIEW

File: backend/api.py [hunk 1]
```diff
@@ -15,162 +15,21 @@ def tester():
...
+def ingredients(username):
+    house_id = service.login(escape(username))
+    if house_id == -1:
+        return jsonify({"id": -2, "name": "input correct username"})
+    
+    ingredients_list = service.houseingre(house_id)
+    if ingredients_list == -1:
+        return jsonify({"id": -3, "name": "No ingredients found"})
+    
+    return jsonify({"id": 0, "ingredients": ingredients_list})
```
Comment (StewartSmyth) [inline backend/api.py]: I think here you might be able to just return the ingredientsList as it is already in a json format from the service and ideally the structure would be:
[
{"id": a, "ingredient": y},
{"id": b, "ingredient": z},
]
Which I believe is the format the service already returns

Comment (Deevun155) [reply backend/api.py]: You're right, turns out it's not needed at all. The output from the new pr looks like this:

[
  {
    "id": 1,
    "name": "Rice"
  },
  {
    "id": 2,
    "name": "Tomato"
  },
  {
    "id": 3,
    "name": "Garlic"
  },
  {
    "id": 4,
    "name": "Cheese"
  },
  {
    "id": 5,
    "name": "Canned Beans"
  },
  {
    "id": 6,
    "name": "Chicken Breast"
  },
  {
    "id": 7,
    "name": "Bread"
  },
  {
    "id": 8,
    "name": "Canned Tuna"
  },
  {
    "id": 9,
    "name": "Spinach"
  },
  {
    "id": 10,
    "name": "Yogurt"
  }
]

**Comment by Deevun155** [comment]
Hang on guys, I think may have messed up a bunch of the service stuff as well. Let me try fix all this and ill do another pr.

**Comment by Deevun155** [comment]
OK, turns out there was just an error with the service_test. I've checked to make sure no other functions are affected by my changes. I'll make another PR to add service_test to CI, but for now this PR should be good to go. Please double check the changes.

**Comment by StewartSmyth** [comment]
Looks good to me

**Review by StewartSmyth — APPROVED** [review]

**Pull request merged**